### PR TITLE
Add #9278 [v97]: Telemetry for tab navigation buttons

### PR DESCRIPTION
**Pull request by nishant2718**
# Overview

PR addresses [this JIRA ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-3263) and #9278. Data Review notes will be posted in the comments soon.

### CODE REVIEW

File: Client/Frontend/Browser/Tab.swift [hunk 1]
```diff
@@ -830,6 +832,34 @@ class Tab: NSObject {
...
 
+extension Tab: UIGestureRecognizerDelegate {
+    // This prevents the recognition of one gesture recognizer from blocking another
+    func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        return true
+    }
+    
+    func configureEdgeSwipeGestureRecognizers() {
+        guard let webView = webView else {
+            os_log(.debug, "Tab's edge swipe gesture recognizer was never added. This will affect Tab navigation telemetry!")
```
Comment (lmarceau) [inline Client/Frontend/Browser/Tab.swift:843]: `os_log` is considered a [legacy logging symbols](https://developer.apple.com/documentation/os/logging/legacy_logging_symbols). When we start using iOS 14 and up we'll be able to use the new os logging API. I would suggest adding a wrapper on top of the unified logging system so we only have to change it in one place when we'll be able to.

File: Client/Frontend/Browser/Tab.swift [hunk 2]
```diff
@@ -830,6 +832,34 @@ class Tab: NSObject {
...
+        }
+        
+        let swipeGesture = UIScreenEdgePanGestureRecognizer(target: self, action: #selector(handleEdgeSwipeTabNavigation(_:)))
+        swipeGesture.edges = .left
+        swipeGesture.delegate = self
+        webView.addGestureRecognizer(swipeGesture)
+    }
+    
+    @objc func handleEdgeSwipeTabNavigation(_ sender: UIScreenEdgePanGestureRecognizer) {
+        if let webView = webView {
```
Comment (lmarceau) [inline Client/Frontend/Browser/Tab.swift:854]: Maybe use `guard let webView = webView else { return }` to reduce arrow-code (`if` in another `if`)? Or combine the two `if` together and put that on multiple lines?

**Comment by nishant2718** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

_1) What questions will you answer with this data?_

It answers how our users navigate within tabs. 

_2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?_

This provides information about user behavior on navigation within tabs. 

_3) What alternative methods did you consider to answer these questions? Why were they not sufficient?_

Current methods have not previously recorded this type of telemetry.

_4) Can current instrumentation answer these questions?_

No.

_5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki._

_**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**_

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>Did a user tap/long press the back button in the toolbar?</td>
    <td>Category 2</td>
    <td>#9278</td>
  </tr>
 <tr>
    <td>Did a user tap/long press the forward button in the toolbar?</td>
    <td>Category 2</td>
    <td>#9278</td>
  </tr>
 <tr>
    <td>Did a user tap the refresh button in the toolbar?</td>
    <td>Category 2</td>
    <td>#9278</td>
  </tr>
 <tr>
    <td>Did a user navigate to a previous link in tab history but swiping from the left edge of the screen?</td>
    <td>Category 2</td>
    <td>#9278</td>
  </tr>
</table>

_6) How long will this data be collected?  Choose one of the following:_

Until 2022-07-01.

_7) What populations will you measure?_

All release channels and locales.

_8) If this data collection is default on, what is the opt-out mechanism for users?_

Users can opt out of of data collection by disabling Usage and technical data from _Settings → Send Usage Data_.

_9) Please provide a general description of how you will analyze this data._

Glean 

_10) Where do you intend to share the results of your analysis?_

Only on Glean, mobile teams and BD have internal access.

_12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?_

No.

File: Client/Frontend/Browser/Tab.swift (hunk 1 above)
Comment (nishant2718) [reply Client/Frontend/Browser/Tab.swift:843]: Great catch! Hmm given that we're iOS13, and we already have a robust logging solution (that would be good to reimplement with iOS14's changes to cut our dependency on that third party lib), maybe we can use the current log solution for now?

**Comment by nishant2718** [comment]
@dnarcese for data review

**Review by lmarceau — APPROVED** [review]

**Comment by dnarcese** [comment]
r+
# Data Review Form

_1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?_ 

Yes, telemetry is documented at https://dictionary.telemetry.mozilla.org/apps/firefox_ios

_2) Is there a control mechanism that allows the user to turn the data collection on and off?_ 

Yes, Firefox iOS provides data controls in the app settings

_3) If the request is for permanent data collection, is there someone who will monitor the data over time?_

This should be re-reviewed w/ Ashley Thomas every 6mo

_4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?_

Type 2

_5) Is the data collection request for default-on or default-off?_

Default on

_6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?_

No

_7) Is the data collection covered by the existing Firefox privacy notice?_ 

Yes

_8) Does the data collection use a third-party collection tool?_ 

No

**Pull request merged**